app/assets/javascripts/discourse/app/widgets/post.js

```diff
@@ -495,7 +495,11 @@ createWidget("post-contents", {
 
     result = result.concat(applyDecorators(this, "after-cooked", attrs, state));
 
-    if (attrs.cooked_hidden) {
+    if (
+      attrs.cooked_hidden &&
+      (this.currentUser?.trust_level === 4 ||
+        attrs.user_id === this.currentUser?.id)
```

Comment:
this is a super small nit, but I would reverse these for consistency:

```js
(this.currentUser?.trust_level === 4 ||
        this.currentUser?.id === attrs.user_id)
```

You may also want to use `isLeader` or `trustLevel === 4` rather than using `trust_level` directly:

https://github.com/discourse/discourse/blob/24ec06ff85c7acbad9621092b5e50eec2ede7b83/app/assets/javascripts/discourse/app/models/user.js#L361-L372